cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/goals/Goal.java

```diff
@@ -34,6 +34,25 @@
  */
 @InterfaceStability.Evolving
 public interface Goal extends Configurable {
+  /**
+   * enum to categorize various types of goals in system
+   * Presently, we have Hard or Soft goals. This might change
+   * in future
+   */
+  public enum GoalType {
```

Comment:
At this point it is not quite clear to me whether we want to just split the `goals` configuration to `hard.goals` or `soft.goals` or just hard code the goal type. The good thing about having two lists is that user can switch the goals between hard and soft more flexibly without code change. If we adopt the `hard.goals` and `soft.goals` configurations, we would not want to add this to the goal interface.

From the UI perspective, it seems this is not a hard requirement, can we just put this aside and add it later?